Tests/SonarScanner.MSBuild.PreProcessor.Test/Caching/CachedDownloaderTests.cs

```diff
@@ -227,21 +227,22 @@ public async Task DownloadFileAsync_ValidFileCached_Succeeds()
     }
 
     [TestMethod]
-    public async Task DownloadFileAsync_InvalidFileCached_ReturnsCacheFailure()
+    public async Task DownloadFileAsync_InvalidFileCached_DeletesAndRedownloads()
     {
         runtime.File.Exists(DownloadFilePath).Returns(true);
-        checksum.ComputeHash(null).ReturnsForAnyArgs("someOtherHash");
+        checksum.ComputeHash(null).ReturnsForAnyArgs(x => "someOtherHash", x => ExpectedSha);
 
         var result = await ExecuteDownloadFileAsync(new MemoryStream(downloadContentArray));
 
-        result.Should().BeOfType<DownloadError>().Which.Message
-            .Should().Be("The checksum of the downloaded file does not match the expected checksum.");
-        runtime.File.DidNotReceiveWithAnyArgs().Create(null);
-        runtime.File.DidNotReceiveWithAnyArgs().Move(null, null);
+        result.Should().BeOfType<DownloadSuccess>().Which.FilePath.Should().Be(DownloadFilePath);
+        runtime.File.Received(1).Create(TempFilePath);
+        runtime.File.Received(1).Move(TempFilePath, DownloadFilePath);
         runtime.Logger.DebugMessages.Should().BeEquivalentTo(
             $"The file was already downloaded from the server and stored at '{DownloadFilePath}'.",
             $"The checksum of the downloaded file is 'someOtherHash' and the expected checksum is '{ExpectedSha}'.",
-            $"Deleting file '{DownloadFilePath}'.");
+            $"Deleting file '{DownloadFilePath}'.",
```

Comment:
In both tests it's possible I've messed up the paths - please take a look at the mock setups mostly.